subprojects/packagefiles/c-ares/meson.build

```diff
@@ -0,0 +1,230 @@
+  'src/lib/ares_version.c',
+  'src/lib/bitncmp.c',
+  'src/lib/inet_net_pton.c',
+  'src/lib/inet_ntop.c',
+  'src/lib/windows_port.c')
+
+cares_headers = files(
+  'src/lib/ares__htable.h',
+  'src/lib/ares__htable_asvp.h',
+  'src/lib/ares__htable_stvp.h',
+  'src/lib/ares__llist.h',
+  'src/lib/ares__buf.h',
+  'src/lib/ares__slist.h',
+  'src/lib/ares_android.h',
+  'src/lib/ares_data.h',
+  'src/lib/ares_getenv.h',
+  'src/lib/ares_inet_net_pton.h',
+  'src/lib/ares_iphlpapi.h',
+  'src/lib/ares_ipv6.h',
+  'src/lib/ares_nowarn.h',
+  'src/lib/ares_platform.h',
+  'src/lib/ares_private.h',
+  'src/lib/ares_strcasecmp.h',
+  'src/lib/ares_strdup.h',
+  'src/lib/ares_strsplit.h',
+  'src/lib/bitncmp.h',
+  'src/lib/ares_setup.h',
+  'src/lib/setup_once.h')
+
+cares_include_dirs = include_directories('include', 'src/lib')
+
+cc = meson.get_compiler('c')
+cares_args_map = {
+'common': [
+  '-DCARES_BUILDING_LIBRARY',
+
+  cc.has_header('arpa/inet.h') ? '-DHAVE_ARPA_INET_H' : [],
+  cc.has_header('errno.h') ? '-DHAVE_ERRNO_H' : [],
+  cc.has_header('limits.h') ? '-DHAVE_LIMITS_H' : [],
+  cc.has_header('netdb.h') ? '-DHAVE_NETDB_H' : [],
+  cc.has_header('netinet/in.h') ? '-DHAVE_NETINET_IN_H' : [],
+  cc.has_header('netinet/tcp.h') ? '-DHAVE_NETINET_TCP_H' : [],
+  cc.has_header('stdint.h') ? '-DHAVE_STDINT_H' : [],
+  cc.has_header('strings.h') ? '-DHAVE_STRINGS_H' : [],
+  cc.has_header('sys/ioctl.h') ? '-DHAVE_SYS_IOCTL_H' : [],
+  cc.has_header('sys/random.h') ? '-DHAVE_SYS_RANDOM_H' : [],
+  cc.has_header('sys/socket.h') ? '-DHAVE_SYS_SOCKET_H' : [],
+  cc.has_header('sys/stat.h') ? '-DHAVE_SYS_STAT_H' : [],
+  cc.has_header('sys/time.h') ? '-DHAVE_SYS_TIME_H' : [],
+  (cc.has_header('sys/time.h') and cc.has_header('time.h')) ? '-DTIME_WITH_SYS_TIME' : [],
+  cc.has_header('sys/types.h') ? '-DHAVE_SYS_TYPES_H' : [],
+  cc.has_header('sys/uio.h') ? '-DHAVE_SYS_UIO_H' : [],
+
+  cc.has_type('struct timeval', prefix: '#include <sys/time.h>') ? '-DHAVE_STRUCT_TIMEVAL' : [],
+  cc.has_type('struct sockaddr_in6', prefix: '#include <sys/socket.h>\n#include <netinet/in.h>') ? '-DHAVE_STRUCT_SOCKADDR_IN6' : [],
+  cc.has_type('struct addrinfo', prefix: '#include <sys/types.h>\n#include <sys/socket.h>\n#include <netdb.h>') ? '-DHAVE_STRUCT_ADDRINFO' : [],
+
+  ## Wants meson 1.3.0
+  # cc.has_define('AF_INET6', prefix: '#include <sys/socket.h>\n#include <netinet/in.h>') ? '-DHAVE_AF_INET6' : [],
+  # cc.has_define('PF_INET6', prefix: '#include <sys/socket.h>\n#include <netinet/in.h>') ? '-DHAVE_PF_INET6' : [],
+  # cc.has_define('O_NONBLOCK', prefix: '#include <fcntl.h>') ? '-DHAVE_FCNTL_O_NONBLOCK' : [],
+
+  cc.has_function('recvfrom', prefix: '#include <sys/types.h>\n#include <sys/socket.h>') ? '-DHAVE_RECVFROM' : [],
+  cc.has_function('writev', prefix: '#include <sys/uio.h>') ? '-DHAVE_WRITEV' : [],
+
+  cc.links('#include <string.h>\nint main() {return (int)strncasecmp("a", "A", 1);}', name: 'strncasecmp') ? '-DHAVE_STRNCASECMP' : [],
```

Comment:
has_header_symbol.

Reply:
These manage to compile somehow, but not link, depending on runtime. So checking for symbol in header is not an option.

Reply:
Oh wait, it's C, maybe it was implicit function declaration

Reply:
Ok, has_header_symbol indeed works in this case. Thanks